packages/placeholder-pdf-lib/src/pdflibAddPlaceholder.js

```diff
@@ -59,24 +71,37 @@ export const pdflibAddPlaceholder = ({
     const placeholder = PDFHexString.of(String.fromCharCode(0).repeat(signatureLength));
 
     // Create a signature dictionary to be referenced in the signature widget.
-    const signatureDict = pdfDoc.context.obj({
+    const signatureDict = doc.context.obj({
         Type: 'Sig',
         Filter: 'Adobe.PPKLite',
         SubFilter: subFilter,
         ByteRange: byteRange,
         Contents: placeholder,
         Reason: PDFString.of(reason),
-        M: PDFString.fromDate(new Date()),
+        M: PDFString.fromDate(signingTime ?? new Date()),
         ContactInfo: PDFString.of(contactInfo),
         Name: PDFString.of(name),
         Location: PDFString.of(location),
-    }, pdfDoc.index);
-    const signatureDictRef = pdfDoc.context.register(signatureDict);
+        Prop_Build: {
+            Filter: {Name: 'Adobe.PPKLite'},
+            App: {Name: 'pdfsign'},
```

Comment:
We should allow the app to set its name that. We are just a helper library.

Reply:
Alright, I will make it an extra parameter to the placeholder function then.
Do you squash all commits when merging or would you like me to rebase everything into 1 commit?

Reply:
I don't usually squash and I usually rebase and reword my commits a lot. But don't feel bothered by that. Whatever is fine for you. The working code is important.